mabwiser/base_mab.py

```diff
@@ -394,3 +402,34 @@ def _get_cold_arm_to_warm_arm(self, cold_arm_to_warm_arm, arm_to_features, dista
                 new_cold_arm_to_warm_arm[cold_arm] = closest_arm
 
         return new_cold_arm_to_warm_arm
+
+    def _reset_arm_to_status(self):
+        self.arm_to_status: Dict[Arm, dict] = {arm: {STATUS_TRAINED: False, STATUS_WARM: False,
+                                                     STATUS_WARM_STARTED_BY: None}
+                                               for arm in self.arms}
+
+    def _set_arms_as_trained(self, arms: Optional[List[Arm]] = None, decisions: Optional[np.ndarray] = None,
+                             is_partial: bool = True):
+        """Sets the given arms as trained.
+
+        Uses the ``arms`` if provided, otherwise calculates the arms from the ``decisions``.
+        Either ``arms`` or ``decisions`` should be provided.
+        """
+        # Check args
+        if arms is None and decisions is None:
+            raise ValueError("Either ``arms`` or ``decisions`` should be specified.")
+
+        # Calculate arms from decisions if ``arms`` is ``None``
+        if arms is None:
+            # Get list of arms in decisions
+            arms = np.unique(decisions).tolist()
+
+        for arm in self.arms:
+            if arm in arms:
+                # All system arms are now trained
+                self.arm_to_status[arm][STATUS_TRAINED] = True
+
+                # If fitting from scratch, arm is no longer warm started
+                if not is_partial:
```

Comment:
Thru this logic, we can now tell when an arm has only been partial fit after a warm start.